src/device/r4300/new_dynarec/arm/linkage_arm.S

```diff
@@ -70,84 +69,46 @@
-
-    dynarec_local                   = extra_memory                      + 33554432
-    stack_space                     = dynarec_local                     + 0  /* Stack space used by dynarec to push/pop caller-saved register (r0-r3, r12) */
-    free_space                      = dynarec_local                     + 20 /* Free space (8 bytes) */
-    saved_context                   = dynarec_local                     + 28
-    g_dev_r4300_cp0_next_interrupt  = dynarec_local                     + 64
-    cycle_count                     = g_dev_r4300_cp0_next_interrupt    + 4
-    last_count                      = cycle_count                       + 4
-    pending_exception               = last_count                        + 4
-    pcaddr                          = pending_exception                 + 4
-    g_dev_r4300_stop                = pcaddr                            + 4
-    invc_ptr                        = g_dev_r4300_stop                  + 4
-    g_dev_r4300_address             = invc_ptr                          + 4
-    readmem_dword                   = g_dev_r4300_address               + 4
-    g_dev_r4300_wdword              = readmem_dword                     + 8
-    g_dev_r4300_wword               = g_dev_r4300_wdword                + 8
-    g_dev_r4300_whword              = g_dev_r4300_wword                 + 4
-    g_dev_r4300_wbyte               = g_dev_r4300_whword                + 2 /* 1 byte free */
-    g_dev_r4300_cp1_fcr0            = g_dev_r4300_whword                + 4
-    g_dev_r4300_cp1_fcr31           = g_dev_r4300_cp1_fcr0              + 4
-    g_dev_r4300_regs                = g_dev_r4300_cp1_fcr31             + 4
-    g_dev_r4300_hi                  = g_dev_r4300_regs                  + 256
-    g_dev_r4300_lo                  = g_dev_r4300_hi                    + 8
-    g_dev_r4300_cp0_regs            = g_dev_r4300_lo                    + 8
-    g_dev_r4300_cp1_regs_simple     = g_dev_r4300_cp0_regs              + 128
-    g_dev_r4300_cp1_regs_double     = g_dev_r4300_cp1_regs_simple       + 128
-    rounding_modes                  = g_dev_r4300_cp1_regs_double       + 128
-    branch_target                   = rounding_modes                    + 16
-    g_dev_r4300_pc                  = branch_target                     + 4
-    fake_pc                         = g_dev_r4300_pc                    + 4
-    ram_offset                      = fake_pc                           + 132
-    mini_ht                         = ram_offset                        + 4
-    restore_candidate               = mini_ht                           + 256
-    memory_map                      = restore_candidate                 + 512
-
-END_SECTION
+.global g_dev
+
+g_dev_r4300_new_dynarec_hot_state_dynarec_local = (g_dev + offsetof_struct_device_r4300_core + offsetof_struct_r4300_core_new_dynarec_hot_state + offsetof_struct_new_dynarec_hot_state_dynarec_local)
```

Comment:
Oups I didn't ! Will fix later :) That should indeed explain the remaining compiling error.